### PR TITLE
feat: obc-wind updates and add obc-current instrument

### PR DESCRIPTION
**Pull request by talpitoo**
## `obc-wind` changes

**New API** (backward compatible; defaults keep prior visuals apart from the face updates below):

- `visualization: 'histogram' | 'forceGraphics'` (default `histogram`) — force-graphics renders the new streak-field underlay instead of the histogram.
- `priority: 'regular' | 'enhanced'` — enhanced recolors the visualization to `--instrument-enhanced-tertiary-color` and the wind arrow to `--instrument-enhanced-secondary-color` (same variables `obc-compass` uses).

**Face updates to match the 6.1 design** (existing wind baselines regenerated deliberately):

- NSEW labels on (inside, flush against the inner ring) — the design shows them on every variant.
- Vessel size 100 → 160 units (`VesselImageSize.medium`), per the design.
- Arrow placement measured from the Figma variants (effective tip radius = `windSymbolRadius × scaleWindIcon`, since the scale group wraps the translate).

The old values were PR #864 "visual-tuning starting points"; no functional reason to keep them was found. In the design, tip-at-center applies to the small/arrow type only — the large/track arrow stays at the inner ring.

**Unchanged by request:** the histogram rendering/data handling itself (the new Figma histogram look — thin fanning lines — is deferred until the data format is confirmed); only its color becomes priority-aware.

## New `obc-current` (`src/navigation-instruments/current/`)

Thin wrapper over `obc-watch`, analogous to `obc-wind`, reusing the existing vessels and `current-0…4` chevron icons:

- `type: 'vessel' | 'direction'` — vessel: double-ring track face, 160-unit vessel rotated by `vesselHeadingDeg`, peripheral chevron icon anchored at radius 116 (tips at the inner ring, inside the track band) rotating with the flow; direction: single-ring face with a large centered chevron glyph (24-unit icon × 7).
- `currentSpeed` (0–4 = number of chevrons, rounded + clamped; `obc-compass` semantics), `currentFromDirection`, `priority`, `vesselImage`, `vesselHeadingDeg`.
- Marked `@experimental`.

## Shared infrastructure

- **`watch/force-pattern.ts`** — parametric reproduction of the exact Figma pattern tiles (read from the exported mask SVGs, no raster assets):
  - wind: 24-unit tile, one 4×24 rect rotated −45° with an alpha gradient (transparent tail → opaque head); tile's native direction is *from 315°*, so the group rotates by `fromDirection − 315` (byte-identical to the design at 315°).
  - current: 48-unit tile, full-square vertical gradient (soft-edged bands, phase offset 104) rotated by `fromDirection`.
  - Pattern square = 256 × patternScale, clipped to the disc; scales measured per usage: wind 0.875 (large) / 1.28125 (medium+small), current 1.39 (vessel) / 2.0 (direction). Both patterns rotate with direction only — the design defines no speed modulation.
- **Centered current icon** — `renderCurrentCentered()` in `environment.ts` plus `currentIconCentered` / `scaleCurrentIcon` on `obc-watch` (mirrors `scaleWindIcon`); the peripheral path is untouched (compass + watch visual suites verified pixel-identical).

## Architecture note

Per the maintainer decision during design review, the pattern renders as an **instrument-local underlay SVG** (stacked behind `obc-watch`, like the wind histogram) rather than inside `watch.ts`; the tile geometry lives in the shared helper so both instruments use one source.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary

- **New Features**
  - Added the **Current** navigation instrument (direction/strength rendering).
  - Added **centered current icon** support in the watch instrument (toggle + scaling).
  - Added **Wind force-graphics** visualization mode, with **regular/enhanced priority** styling.
  - Expanded **Storybook** examples for Current and Wind.
- **Bug Fixes**
  - Improved current and wind indicator sizing, placement, rotation, and color/pattern rendering consistency.
- **Tests**
  - Added automated coverage for current speed clamping/edge cases and SVG rendering for current/wind force patterns and placements.

### CODE REVIEW

File: packages/openbridge-webcomponents/src/navigation-instruments/current/current.stories.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import type {Meta, StoryObj} from '@storybook/web-components-vite';
+import {ObcCurrent, CurrentType} from './current.js';
+import './current.js';
+import {Priority} from '../types.js';
+import {widthDecorator} from '../../storybook-util.js';
+import {topVessels} from '../watch/vessels/storybook-helper.js';
+
+const meta: Meta<typeof ObcCurrent> = {
+  title: 'Instruments/Current',
+  tags: ['autodocs', '6.0'],
```
[the remark this answers is not in the document]
Comment (talpitoo) [reply packages/openbridge-webcomponents/src/navigation-instruments/current/current.stories.ts]: Done in 1dd63686 — with `['autodocs', 'wip']` rather than bare `['alpha']`: position-deviation (the newest @experimental navigation instrument) established `['autodocs', 'wip']` for in-development instruments, and dropping autodocs would remove the generated docs page.

File: packages/openbridge-webcomponents/src/navigation-instruments/current/current.stories.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  args: {type: CurrentType.direction},
+};
+
+export const VesselEnhanced: Story = {
+  args: {priority: Priority.enhanced},
+};
+
+export const DirectionEnhanced: Story = {
+  args: {type: CurrentType.direction, priority: Priority.enhanced},
+};
```
[the remark this answers is not in the document]
Comment (talpitoo) [reply packages/openbridge-webcomponents/src/navigation-instruments/current/current.stories.ts]: Done in 1dd63686 — renamed `Vessel` → `Default` (its args were already the default configuration, so the render and the baseline are pixel-identical; baseline file renamed accordingly). `Direction` and the enhanced variants stay as-is.

File: packages/openbridge-webcomponents/src/navigation-instruments/current/current.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+   * The current strength as a chevron bucket (0–4 = number of chevrons);
+   * rounded and clamped. `null` hides the icon.
+   * @availableWhen currentFromDirection!=null
+   */
+  @property({type: Number}) currentSpeed: number | null = null;
+  /**
+   * The direction the current is coming from in degrees.
+   * @availableWhen currentSpeed!=null
+   */
+  @property({type: Number}) currentFromDirection: number | null = null;
```
[the remark this answers is not in the document]
Comment (talpitoo) [reply packages/openbridge-webcomponents/src/navigation-instruments/current/current.ts:89]: Done in 1dd63686 — removed `@availableWhen currentSpeed!=null` from `currentFromDirection`; the force pattern renders from direction alone, so per the multi-path rule the property must not be annotated. `currentSpeed`'s own `@availableWhen currentFromDirection!=null` stays: its only effect (the chevron icon) genuinely requires a direction.

File: packages/openbridge-webcomponents/src/navigation-instruments/watch/watch.ts [hunk 4]
```diff
@@ -285,6 +273,17 @@ export class ObcWatch extends LitElement {
...
+   * Render the current icon centered on the face (direction-type layout)
+   * instead of at `currentSymbolRadius` on the periphery.
+   * @availableWhen current!=null && currentFromDirectionDeg!=null
+   */
+  @property({type: Boolean}) currentIconCentered: boolean = false;
+  /**
+   * Scale factor for the centered current icon.
+   * @availableWhen currentIconCentered==true
+   */
+  @property({type: Number}) scaleCurrentIcon: number = 1;
```
[the remark this answers is not in the document]
Comment (talpitoo) [reply packages/openbridge-webcomponents/src/navigation-instruments/watch/watch.ts:302]: Rejected — the annotation follows existing practice and is accurate against the render logic: `currentIconCentered` only has an observable effect when an icon renders at all (`current != null && currentFromDirectionDeg != null`). An annotated property that also gates another is established in this codebase: compass's `autoAtHeadingSetpoint` carries `@availableWhen headingSetpoint!=null` while appearing in the conditions of `atHeadingSetpoint` and `autoAtHeadingSetpointDeadband`. The "never annotate the gate itself" rule means a gate doesn't get annotated with its dependents' condition — not that a gate can't have its own upstream dependency.

**Review by ulrik-jo — APPROVED** [review]
